### PR TITLE
This avoids a CF allocation on data fetcher in the main case where there is no exception

### PR DESCRIPTION
**Pull request by bbakerman**

### CODE REVIEW

**Comment by dfa1** [comment]
@bbakerman what about using a single `.handle`?

```
        return fetchedValue
                .handle((result, exception) -> {
                    fetchCtx.onCompleted(result, exception);
                    if (exception != null) {
                        result = handleFetchingException(executionContext, dataFetchingEnvironment.get(), exception);
                    }
                    return unboxPossibleDataFetcherResult(executionContext, parameters, result);
                });
```

**Comment by bbakerman** [comment]
> What about using a single .handle?
Ahh I tried it - it wont work

The error handler returns `CF<U>`  while `.handle()` needs to a `<U>` so we need to compose afterwards - I dont think there is any saving to be made here

File: src/main/java/graphql/execution/ExecutionStrategy.java [hunk 1]
```diff
@@ -363,9 +364,9 @@ private void addExtensionsIfPresent(ExecutionContext executionContext, DataFetch
         }
     }
 
-    protected <T> CompletableFuture<T> handleFetchingException(ExecutionContext executionContext,
-                                                               DataFetchingEnvironment environment,
-                                                               Throwable e) {
+    protected <T> CompletableFuture<T> handleFetchingException(
```
Comment (bbakerman) [inline src/main/java/graphql/execution/ExecutionStrategy.java:367]: never used

File: src/main/java/graphql/execution/ExecutionStrategy.java [hunk 2]
```diff
@@ -714,19 +715,6 @@ private Object handleCoercionProblem(ExecutionContext context, ExecutionStrategy
...
-    /**
-     * Converts an object that is known to should be an Iterable into one
-     *
-     * @param result the result object
-     *
-     * @return an Iterable from that object
-     *
-     * @throws java.lang.ClassCastException if it's not an Iterable
-     */
-    protected Iterable<Object> toIterable(Object result) {
```
Comment (bbakerman) [inline src/main/java/graphql/execution/ExecutionStrategy.java:726]: never used

Comment (bbakerman) [inline src/main/java/graphql/execution/ExecutionStrategy.java:726]: technically a breaking change BUT a low risk one

**Pull request merged**